src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework.UnitTests/Mocks/TestHandlerWithLanguageProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.CommonLanguageServerProtocol.Framework.UnitTests;
+
+internal class TestHandlerWithLanguageProvider : IHandlerProvider
+{
+    private readonly IEnumerable<(RequestHandlerMetadata metadata, IMethodHandler provider)> _providers;
+
+    public TestHandlerWithLanguageProvider(IEnumerable<(RequestHandlerMetadata metadata, IMethodHandler provider)> providers)
```

Comment:
this is strange, but i gess ok since this is tests. normally this would be an `IEnumerable<Lazy<IMethodHandler, RequestHandlerMetadata>>`.